sideline-eglot.el

```diff
@@ -52,17 +60,39 @@
      ,@body
      (advice-remove 'y-or-n-p #'always)))
 
-(defvar-local sideline-eglot--ht-candidates nil
+(defvar sideline-eglot--ht-candidates nil
   "Holds candidates.")
 
+(defvar sideline-eglot--callback)
+
 (defun sideline-eglot--async-candidates (callback &rest _)
-  "Request eglot's candidates."
-  (if sideline-eglot--ht-candidates
-      (ht-clear sideline-eglot--ht-candidates)
-    (setq sideline-eglot--ht-candidates (ht-create)))
-  (dolist (row (ignore-errors (eglot-code-actions (point))))
-    (ht-set sideline-eglot--ht-candidates (cl-getf row :title) row))
-  (funcall callback (ht-keys sideline-eglot--ht-candidates)))
+  "Request eglot's CALLBACK candidates."
+  (setq sideline-eglot--callback callback)
+  (jsonrpc-async-request
+   (eglot-current-server)
+   :textDocument/codeAction
+   (list :textDocument (eglot--TextDocumentIdentifier)
+         :range (list :start (eglot--pos-to-lsp-position (point))
+                      :end (eglot--pos-to-lsp-position nil))
+         :context
+         `(:diagnostics
+           [,@(cl-loop for diag in (flymake-diagnostics (point) nil)
+                       when (cdr (assoc 'eglot-lsp-diag
+                                        (eglot--diag-data diag)))
+                       collect it)]))
+   :success-fn
+   (lambda (resp)
+     (let ((actions (append resp nil)))
+       (if sideline-eglot--ht-candidates
+           (ht-clear sideline-eglot--ht-candidates)
+         (setq sideline-eglot--ht-candidates (ht-create)))
+       (dolist (row actions)
+         (ht-set sideline-eglot--ht-candidates
+                 (concat sideline-eglot-code-actions-prefix
+                         (cl-getf row :title))
+                 row))
+       (funcall sideline-eglot--callback (ht-keys sideline-eglot--ht-candidates))))
```

Comment:
In the lambda, you will lose buffer local variables `sideline-eglot--callback` and `sideline-eglot--ht-candidates` since you are inside the `*temp*` buffer.